### PR TITLE
Reducing some code by cleaning up some logic

### PR DESCRIPTION
**Pull request by joesho112358**
Reading through, I noticed some things could be combined to reduce reading and indentation.

### CODE REVIEW

File: cirq-core/cirq/circuits/circuit.py [hunk 1]
```diff
@@ -573,14 +573,12 @@ def enqueue_next(qubit: 'cirq.Qid', moment: int) -> None:
...
+                if continue_past:
                     enqueue_next(q, cur_moment + 1)
-            else:
-                for q in cur_op.qubits:
-                    if q in active:
-                        end_frontier[q] = cur_moment
-                        active.remove(q)
+                elif q in active:
+                    end_frontier[q] = cur_moment
+                    active.remove(q)
```
Comment (viathor) [inline cirq-core/cirq/circuits/circuit.py:581]: This is shorter, but tests the condition on each iteration. I think the previous code is slightly better.

**Review by viathor — CHANGES_REQUESTED** [review]
Thank you for the PR! I agree two of the changes improve the code. Before merging, I would like to ask you to revert one of them to avoid repeated checks in a loop, see below.

File: cirq-core/cirq/circuits/circuit.py (hunk 1 above)
Comment (joesho112358) [reply cirq-core/cirq/circuits/circuit.py]: makes sense to me! pushed.

**Review by viathor — APPROVED** [review]
Thank you!

**Pull request merged**